src/components/utils/useLayer.ts

```diff
@@ -50,12 +63,20 @@ export function useLayer({
         onClose,
         contentRefs,
         enabled,
+        setAllowedOpen,
     ]);
 
     React.useEffect(() => {
         if (open && enabled) {
             const layerConfig = layerConfigRef.current;
-            layerManager.add(layerConfig);
+            layerManager
+                .add(layerConfig)
+                .then(() => {
+                    setAllowedOpen(true);
+                })
+                .catch(() => {
+                    setAllowedOpen(false);
+                });
 
             return () => {
                 layerManager.remove(layerConfig);
```

Comment:
Do we really need `layerConfig.onRemove` callback?

```suggestion
                layerManager.remove(layerConfig);
                setAllowedOpen(false);
```

Reply:
I thought about this, but current config can be removed from layerManager independently from `useLayer` (e.g. click outside popup). This way discussed effect cleanup will not be called and `allowedOpen` will be stuck at `true`

Reply:
You are right. Fixed